guides/source/asset_pipeline.md

```diff
@@ -3,561 +3,545 @@
+    <!DOCTYPE html>
+    <html>
+      <head>
+        <title>MyApp</title>
+        <%= stylesheet_link_tag "main" %>
+      </head>
+      <body>
+        <%= yield %>
+        <%= javascript_include_tag "main" %>
+      </body>
+    </html>
+    ```
 
-```ruby
-config.assets.unknown_asset_fallback = false
-```
+    This layout includes the `main.css` stylesheet and `main.js` JavaScript file
+    in your application.
 
-If "asset fallback" is enabled then when an asset cannot be found the path will be
-output instead and no error raised. The asset fallback behavior is disabled by default.
+4. Start your Rails server:
 
-### Turning Digests Off
+    ```bash
+    $ bin/rails server
+    ```
 
-You can turn off digests by updating `config/environments/development.rb` to
-include:
+    This command starts your Rails server so you can view your application in a
+    web browser.
 
-```ruby
-config.assets.digest = false
-```
+5. Preview your application:
 
-When this option is true, digests will be generated for asset URLs.
+    Open your web browser and navigate to `http://localhost:3000`. You should
+    see your Rails application with the included assets.
 
-### Turning Source Maps On
+### Development
 
-You can turn on source maps by updating `config/environments/development.rb` to
-include:
+In development, Rails and Propshaft are configured for flexibility, allowing
+rapid iteration without manual intervention. This setup makes it easier to
+manage assets as they change frequently during development.
 
-```ruby
-config.assets.debug = true
-```
+#### No Caching in Development
 
-When debug mode is on, Sprockets will generate a Source Map for each asset. This
-allows you to debug each file individually in your browser's developer tools.
+In development, Rails is configured to bypass asset caching. This means that
+when you modify assets (e.g., CSS, JavaScript), Rails will serve the most
+up-to-date version directly from the file system. There's no need to worry about
+versioning or file renaming because caching is skipped entirely. Browsers will
+automatically pull in the latest version each time you reload the page.
 
-Assets are compiled and cached on the first request after the server is started.
-Sprockets sets a `must-revalidate` Cache-Control HTTP header to reduce request
-overhead on subsequent requests - on these the browser gets a 304 (Not Modified)
-response.
+#### Automatic Reloading of Assets
 
-If any of the files in the manifest change between requests, the server
-responds with a new compiled file.
+If you're using Propshaft on its own, it automatically checks for updates to
+assets like JavaScript, CSS, or images with every request. This means you can
+edit these files, reload the browser, and instantly see the changes without
+needing to restart the Rails server.
 
-In Production
--------------
+When using modern JavaScript bundlers such as
+[esbuild](https://esbuild.github.io/) or [Webpack](https://webpack.js.org/)
```

Comment:
```suggestion
When using JavaScript bundlers
```

"modern" is subjective (Is Webpack considered modern by everyone? For how long would it still be the case?). It does not seem required to get the point across.